### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by brtivedi5**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Completed assignment-2, lab4-6
## What did you learn from the changes you have made?
I have learnt lab works and deep learning models.
## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
There can be more approaches, this one was the best i could find
## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Yes, there are cumbersome reading.
## How were these changes tested?
Inside of model with jyupiter cells
## A reference to a related issue in your repository (if applicable)
NA
## Checklist
- [ ] I can confirm that my changes are working as intended

## Summary by Sourcery

Implements a neural network using Keras to classify handwritten digits from the MNIST dataset. It includes data loading, preprocessing, model building, training, and evaluation. It also explores the impact of different optimizers and initialization schemes on model performance.

New Features:
- Implements a feed forward neural network with Keras.
- Explores the impact of different optimizers and initialization schemes on model performance.
- Adds code to visualize the predictions the model is making on the test set, and identifies the misclassified samples.
- Adds code to explore the parameters of a class by using the question mark

Tests:
- Adds code to split the data into training and test sets, and evaluates the model on the test set.

### CODE REVIEW

**Review by x-rojas-io — APPROVED** [review]
Hi!

Thank you for submitting your DL assignments.

Your A2 and Labs 4-6 are now Complete.

Thanks,
Nestor [LS]
